### PR TITLE
Daily

### PR DESCRIPTION
**Pull request by elguapogordo**
daily story c/w
daily page hit checks to see if there is already a message
message is generated internally through the generator now
names are also generated internally now

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -189,7 +189,7 @@ p{
 
 .daily-subtitle-neo{
   color: white;
-  font-size: 10px;
+  font-size: 12px;
```
Comment (elguapogordo) [inline app/assets/stylesheets/application.css:192]: 10 is a little small

File: app/controllers/daily_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class DailyController < ApplicationController
   def index
-    @events = NeoSearch.new.daily_neos
+    @message = NeoSearch.new.daily_message
```
Comment (elguapogordo) [inline app/controllers/daily_controller.rb:3]: passing daily message to view

File: app/poros/neo_search.rb [hunk 3]
```diff
@@ -8,33 +8,42 @@ def birthday_data(birthdate)
     NeoService.new.birthday(birthdate)
   end
 
+  def daily_message
+    unless DailyMessage.where(date: Date.today.to_s).empty?
+      return @message = DailyMessage.where(date: Date.today.to_s).first
+    else
+      daily_neos
+    end
+  end
```
Comment (elguapogordo) [inline app/poros/neo_search.rb:17]: this checks to see if there is already a daily message and uses it if there is one. if not, then it generates one

File: app/poros/neo_search.rb [hunk 4]
```diff
@@ -8,33 +8,42 @@ def birthday_data(birthdate)
...
+  def daily_neos
+    message = new_daily_message
+    names = Name.new
+    events = NeoService.new.daily
+      events.map do |neo|
+      harbinger = message.harbingers.create(harbinger_params(neo, names.get_name))
+      harbinger.events.create(event_params(neo))
+    end
+    message
+  end
```
Comment (elguapogordo) [inline app/poros/neo_search.rb:28]: this brings in names from the names list now and ties the daily message to harbingers with corresponding events.

File: app/poros/neo_search.rb [hunk 5]
```diff
@@ -8,33 +8,42 @@ def birthday_data(birthdate)
...
-        harbinger.events.create(
-          event_date: neo[:event_date],
-          relative_velocity_mph: neo[:relative_velocity].to_f.round(2),
-          lunar_distance: neo[:lunar_distance].to_f.round(2),
-        )
-      end
-    # end
+  def new_daily_message
+    DailyMessage.create(description: Scope.new.description.upcase, date: Date.today.to_s)
+  end
```
Comment (elguapogordo) [inline app/poros/neo_search.rb:36]: daily message is generated here instead of being hard coded

File: app/views/daily/index.html.erb [hunk 6]
```diff
@@ -5,19 +5,19 @@
...
           <div class="daily-harbinger-info">
-            <li class="name">NAME: <%= event.harbinger.name.upcase %></li><br>
-            <li class="velocity">RELATIVE VELOCITY (MPH): <%= event.relative_velocity_mph %></li><br>
-            <li class="distance">LUNAR DISTANCE: <%= event.lunar_distance %></li>
+            <li class="name">NAME: <%= harbinger.name.upcase %></li><br>
+            <li class="velocity">RELATIVE VELOCITY (MPH): <%= harbinger.events.last.relative_velocity_mph %></li><br>
+            <li class="distance">LUNAR DISTANCE: <%= harbinger.events.last.lunar_distance %></li>
           </div>
         </ul>
       <% end %>
```
Comment (elguapogordo) [inline app/views/daily/index.html.erb:23]: changes in the view to reflect bringing in @message instead of @events

File: app/poros/neo_search.rb [hunk 7]
```diff
@@ -8,33 +8,42 @@ def birthday_data(birthdate)
     NeoService.new.birthday(birthdate)
   end
 
+  def daily_message
+    unless DailyMessage.where(date: Date.today.to_s).empty?
+      return @message = DailyMessage.where(date: Date.today.to_s).first
```
Comment (Lithnotep) [inline app/poros/neo_search.rb:13]: Like this idea to keep the daily messages the same once they are made.

**Pull request merged**